tests/autoscaler/autoscaler_test.go

```diff
@@ -0,0 +1,199 @@
+package ingress
+
+import (
+	"context"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+
+	"github.com/giantswarm/micrologger"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/sonobuoy-plugin/v5/tests/ctrlclient"
+)
+
+const (
+	helloWorldNamespace      = "default"
+	helloWorldDeploymentName = "helloworld"
+)
+
+func Test_Autoscaler(t *testing.T) {
+	var err error
+
+	ctx := context.Background()
+
+	tcCtrlClient, err := ctrlclient.CreateTCCtrlClient(ctx)
+	if err != nil {
+		t.Fatalf("error creating TC k8s client: %v", err)
+	}
+
+	logger, err := micrologger.New(micrologger.Config{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logger.Debugf(ctx, "Creating %s deployment", helloWorldDeploymentName)
+
+	deployment, err := createDeployment(ctx, tcCtrlClient, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tcCtrlClient.Delete(ctx, deployment)
```

Comment:
I tried using `defer` in tests, but I've experience that it only works when leaving the function using `return`. Calling `Fatal` would exit immediately and the deferred function won't be called.

Reply:
Oh man... That sounds really bad. :disappointed:

Reply:
I pushed an alternative proposal, have a look and let me know your thought, please